### DISCUSSION TITLE
Code quality

### DISCUSSION THREAD

**Discussion by jvictorfranca** [directus/directus#13087] [Archive]
Hi! I was searching for a tool to check Directus code-quality and couldnt found one. I tried passing it throught SonarQube, but the results were not right (many security hotspots on unity tests and css, and sonar couldnt get the unit tests to increase its code coverage)

![image](https://user-images.githubusercontent.com/83782983/166251848-6122b04e-7f2b-4eb8-b3f3-db04737c83f8.png)

I also found that scruntizer has directus, but it seems to be old: https://scrutinizer-ci.com/g/directus/directus/

So i was wondering, is eslint the only code-quality measure for Directus? Thanks so much!

**Comment by rijkvanzanten** [comment]
> So i was wondering, is eslint the only code-quality measure for Directus? Thanks so much!

Nope! We also rely on unit/integration/e2e tests, and CodeQL through GitHub actions to auto-check coverage and perform static analysis 👍🏻
